knit/src/Knit/Argument.hs

```diff
@@ -286,17 +297,18 @@ consumeArguments ac args =
             Nothing -> Left procError
             Just a  -> Right a
 
-isArgPos :: Arg a -> Bool
+isArgPos :: Arg ext a -> Bool
 isArgPos = \case
-    ArgPos _ -> True
+    ArgPos _ _ -> True
     _ -> False
 
 toIrrelevanceError
-  :: [Arg (Value components)]
+  :: [Arg ext (Value components)]
   -> ArgumentError
 toIrrelevanceError = foldMap $ \case
-    ArgPos _ -> mempty { aeIrrelevantPos = 1 }
-    ArgKw key _ -> mempty { aeIrrelevantKeys = Set.singleton key }
+    ArgPos _ _ -> mempty { aeIrrelevantPos = 1 }
+    ArgKw _ key _ -> mempty { aeIrrelevantKeys = Set.singleton key }
+    XArg _ -> mempty
```

Comment:
I don't think that would be a correct way to handle `XArg`. 

1. Add a `XXArg ext ~ Void` constraint here and call `absurd`.
2. Remove the `XArg` constructor altogether.